File: Python/Program/Product.py
class Product:
    def __init__(self, idProduct="", name="", brand="", price=0.0):
        self.idProduct = idProduct  #atribut untuk menampung ID produk
        self.name = name            #atribut untuk menampung nama produk
        self.brand = brand          #atribut untuk menampung brand produk
        self.price = price          #atribut untuk menampung harga produk

    #Getter dan Setter
    def setIdProduct(self, idProduct):
        self.idProduct = idProduct

    def setName(self, name):
        self.name = name

    def setBrand(self, brand):
        self.brand = brand

    def setPrice(self, price):
        self.price = price

    def getIdProduct(self):
        return self.idProduct

    def getName(self):
        return self.name

    def getBrand(self):
        return self.brand

    def getPrice(self):
        return self.price

File: PHP/Program/Product.php
<?php
class Product {
    protected $idProduct; // ID produk
    protected $name; // Nama produk
    protected $brand; // Merek produk
    protected $price; // Harga produk

    // Konstruktor untuk inisialisasi objek produk
    public function __construct($idProduct, $name, $brand, $price) {
        $this->idProduct = $idProduct;
        $this->name = $name;
        $this->brand = $brand;
        $this->price = $price;
    }

    // Method untuk memformat harga ke dalam format Rupiah
    public function formatPrice() {
        return "Rp " . number_format($this->price, 0, ',', '.');
    }

    // Getter untuk mendapatkan ID produk
    public function getIdProduct() {
        return $this->idProduct;
    }

    // Getter untuk mendapatkan nama produk
    public function getName() {
        return $this->name;
    }

    // Getter untuk mendapatkan merek produk
    public function getBrand() {
        return $this->brand;
    }

    // Getter untuk mendapatkan harga produk
    public function getPrice() {
        return $this->price;
    }

    // Setter untuk mengatur ID produk
    public function setIdProduct($idProduct) {
        $this->idProduct = $idProduct;
    }

    // Setter untuk mengatur nama produk
    public function setName($name) {
        $this->name = $name;
    }

    // Setter untuk mengatur merek produk
    public function setBrand($brand) {
        $this->brand = $brand;
    }

    // Setter untuk mengatur harga produk
    public function setPrice($price) {
        $this->price = $price;
    }

    // Method untuk mendapatkan informasi lengkap tentang produk
    public function getInfo() {
        return "ID: {$this->idProduct}, Name: {$this->name}, Brand: {$this->brand}, Price: {$this->formatPrice()}";
    }
}
?>
